src/Common/MainController.h

```diff
@@ -315,7 +315,7 @@ public slots:
 
     QScopedPointer<Geo::IpToLocationResolver> ipToLocationResolver;
 
-    QList<Recorder::JamRecorder> jamRecorders;
+    QList<Recorder::JamRecorder *> jamRecorders;
```

Comment:
> So should I create the (empty) list in the dialog constructor and then run through it deleting the referenced objects in the destructor?

Yes @pljones. The empty list will be created by default, but is necessary delete the referenced objects. I see in your commits you tried a **QList<Recoder::JamRecorder>**, without pointers.  I think this is the best solution, no need to delete the objets manually. But I think in this solution we need a default constructor in JamRecorder, or QList will emit a complicated *template* error.  A third solution is use **QList<QScopedPointer<Recorder::JamRecorder>>**, the scoped pointer will auto delete the object reference. Unfortunally this approach is very verbose, but works.